swift-ts-mode.el

```diff
@@ -262,33 +262,37 @@
    '(
      (function_declaration (simple_identifier) @font-lock-function-name-face)
      ;; TODO: Use custom font face with fallback on default for parameters.
-     (value_argument (simple_identifier) @font-lock-type-face)
+         (value_argument_label (simple_identifier) @font-lock-property-use-face)
      (parameter external_name: (simple_identifier) @font-lock-variable-name-face)
      (parameter name: (simple_identifier) @font-lock-variable-name-face)
      (tuple_type_item name: (simple_identifier) @font-lock-variable-name-face)
      (type_parameter (type_identifier) @font-lock-variable-name-face)
      (inheritance_constraint (identifier (simple_identifier)) @font-lock-variable-name-face)
      (equality_constraint (identifier (simple_identifier)) @font-lock-variable-name-face)
-     (lambda_parameter (simple_identifier) @font-lock-variable-name-face))
+         (lambda_parameter (simple_identifier) @font-lock-variable-name-face)
+         (protocol_function_declaration name: (simple_identifier) @font-lock-function-name-face))
 
    :language 'swift
    :feature 'function
    '((call_expression
       (navigation_expression
        suffix: (navigation_suffix suffix: (simple_identifier) @font-lock-function-call-face)))
 
-     ;; TODO: Which feature does this belong?
-     (navigation_expression
-      suffix: (navigation_suffix suffix: (simple_identifier) @font-lock-property-use-face))
-     
      ((directive) @font-lock-preprocessor-face)
-     (prefix_expression (simple_identifier) @font-lock-function-call-face)
-     (call_expression (simple_identifier) @font-lock-function-call-face))
+         ;; distinguish from dictionary-access which has exact same syntax-tree
+         ;; except [ braces ] inside value_arguments.
+         (call_expression
+          (simple_identifier) @font-lock-function-call-face
+          (call_suffix
+           (value_arguments
+            ["("] @open-paren
+            (_) *
+            [")"])))
+         (macro_invocation (simple_identifier) @font-lock-function-call-face))
```

Comment:
I just added macro_invocation highlighting in #4. I used keyword-face instead, but I guess call-face is more correct. That said, I kinda like that it's highlighted with a different face from normal function calls. E.g.

**@font-lock-function-call-face**:

<img width="332" height="79" alt="Screenshot 2025-09-14 at 12 38 09" src="https://github.com/user-attachments/assets/3572b546-2637-4678-a6cc-258473ad7a6a" />

**@font-lock-keyword-face:**

<img width="332" height="78" alt="Screenshot 2025-09-14 at 12 38 24" src="https://github.com/user-attachments/assets/11818df6-7bd3-4cfc-b5cf-cfaa1b1ea296" />

What do you think?

Reply:
I see that rust-ts-mode is using font-lock-preprocessor-face for macros. Maybe it makes sense to do the same here?

Reply:
That’s ok by me too. It’s by far the least important bit of the PR.